incubator/binding-mqtt-kafka/src/main/java/io/aklivity/zilla/runtime/binding/mqtt/kafka/internal/MqttKafkaConfiguration.java

```diff
@@ -70,29 +70,34 @@ public MqttKafkaConfiguration(
         super(MQTT_KAFKA_CONFIG, config);
     }
 
-    public Supplier<String> sessionId()
+    @FunctionalInterface
+    public interface StringSupplier extends Supplier<String>
     {
-        return () -> SESSION_ID.get(this);
     }
 
-    public Supplier<String> willId()
+    public StringSupplier sessionId()
```

Comment:
This is very close. The return type should be the more general `Supplier<String>` and the strongly typed `StringSupplier` should be `private`. Please see `KAFKA_CLIENT_SASL_SCRAM_NONCE` in `KafkaConfiguration` as an example.